### PR TITLE
implement mark as read feature

### PR DESCRIPTION
**Pull request by aceTmobile**
- allow user to mark a book as finished from the reading list side nav
- Indicate the book as finished in the side nav, including the finished date.
- user can still remove the book from their reading list. Removing a book will reset the finished status (i.e. if they add the book back it will not be finished).
- The Want to Read button should change to Finished.
- e2e test implemented

### CODE REVIEW

File: libs/api/books/src/lib/reading-list.service.ts [hunk 1]
```diff
@@ -28,4 +28,11 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async markAsRead(id: string, body: ReadingListItem): Promise<void> {
+    this.storage.update(list => list.map(item => item.bookId == id
```
Comment (mandarini) [inline libs/api/books/src/lib/reading-list.service.ts]: Linting error here! Please make sure `npm run lint` passes

File: libs/books/feature/src/lib/reading-list/reading-list.component.html [hunk 2]
```diff
@@ -8,9 +8,25 @@
       <div class="reading-list-item--details--author">
         {{ b.authors.join(',') }}
       </div>
+      <div id="finishedDate" *ngIf="b.finishedDate" class="reading-list-item--details--finished-date">
+        finished: {{ b.finishedDate }}
```
Comment (mandarini) [inline libs/books/feature/src/lib/reading-list/reading-list.component.html]: Please make sure this date is in a human-readable format!

**Review by mandarini — CHANGES_REQUESTED** [review]
Please fix the issues here!

**Comment by aceTmobile** [comment]
@mandarini Hi, I made corrections. The mr should be good to go now. Please re-review. thanks

**Comment by mandarini** [comment]
Thanks! You're done!

**Pull request merged**